src/hct_mis_api/apps/payment/migrations/0149_migration.py

```diff
@@ -0,0 +1,285 @@
+    ServiceProvider = apps.get_model("payment", "ServiceProvider")
+    FinancialServiceProvider = apps.get_model("payment", "FinancialServiceProvider")
+    DeliveryMechanismPerPaymentPlan = apps.get_model("payment", "DeliveryMechanismPerPaymentPlan")
+    PaymentVerificationSummary = apps.get_model("payment", "PaymentVerificationSummary")
+    PaymentVerificationPlan = apps.get_model("payment", "PaymentVerificationPlan")
+    PaymentVerification = apps.get_model("payment", "PaymentVerification")
+    TicketComplaintDetails = apps.get_model("grievance", "TicketComplaintDetails")
+    TicketSensitiveDetails = apps.get_model("grievance", "TicketSensitiveDetails")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    content_type_for_payment_plan = ContentType.objects.get_for_model(PaymentPlan)
+    content_type_for_cash_plan = ContentType.objects.get_for_model(CashPlan)
+    content_type_for_payment = ContentType.objects.get_for_model(Payment)
+    content_type_for_payment_record = ContentType.objects.get_for_model(PaymentRecord)
+
+    delivery_type_to_obj = {obj.name: obj for obj in DeliveryMechanism.objects.all()}
+
+    for sp in ServiceProvider.objects.all():
+        if not sp.cash_plans.exists():
+            continue
+
+        if FinancialServiceProvider.objects.filter(vision_vendor_number=sp.vision_id).exists():
+            continue
+
+        delivery_mechanisms = set(sp.cash_plans.all().values_list("delivery_type", flat=True))
+
+        fsp = FinancialServiceProvider.objects.create(
+            name=sp.full_name,
+            vision_vendor_number=sp.vision_id,
+            communication_channel="API",
+            internal_data={
+                "is_cash_assist": True,
+                "business_area": sp.business_area.slug,
+                "country": sp.country,
+                "ca_id": sp.ca_id,
+                "short_name": sp.short_name,
+            },
+        )
+        fsp.delivery_mechanisms.set([delivery_type_to_obj[dt] for dt in delivery_mechanisms])
+
+    fsp_vision_vendor_number_to_obj = {obj.vision_vendor_number: obj for obj in FinancialServiceProvider.objects.all()}
+
+    for cp in CashPlan.objects.all():
+
+        with transaction.atomic():
+
+            if not cp.payment_items.exists():
+                continue
+
+            # get target populations from payment records
+            target_populations = set([item.target_population for item in cp.payment_items.all()])
+            # for each target population create a payment plan within tp.payment_cycle
+            for tp in target_populations:
+                payment_records = cp.payment_items.filter(target_population=tp)
+                first_record = payment_records.first()
+                if first_record.delivery_type:
+                    delivery_mechanism = delivery_type_to_obj[first_record.delivery_type.name]
+                else:
+                    delivery_mechanism = delivery_type_to_obj["Cash"]
```

Comment:
maybe assign `delivery_mechanism = delivery_type_to_obj["Cash"]` to a variable before the loop since it's a const